packages/hardhat-vyper/src/types.ts

```diff
@@ -1,7 +1,13 @@
 export type VyperUserConfig = string | VyperConfig | MultiVyperConfig;
 
+export interface VyperSettings {
+  evmVersion?: string;
+  optimize?: "gas" | "codesize" | "none";
```

Comment:
Perhaps this should be just a `string`. Otherwise, if vyper adds a new `optimize` mode, we would need to release a new version adding it. But OTOH  new modes are not added all the time, and using `string` would mean that wrong values would result in a (vyper) execution error instead of a typing error. So I don't know. What would you prefer @pcaversaccio?

Reply:
Hmm, so there is an issue here. For Vyper versions `<0.3.10`, the `optimize` property was a Boolean: https://docs.vyperlang.org/en/v0.3.9/compiling-a-contract.html#setting-the-target-evm-version (see the PR https://github.com/vyperlang/vyper/pull/3493)). So the type changes depending on the Vyper version (some pseudocode lol):

```
(if evmVersion < 0.3.10) ? (typeof(optimize) == bool) : (typeof(optimize) == string)
```

Reply:
From what I see in the help output of vyper `0.3.9`, there is a `--no-optimize` flag, but there's no `--optimize` one. Does this mean that versions older than 0.3.10 use the optimizer by default? In that case, maybe the behavior should be something like this:

```js
if (typeof optimize === "boolean") {
  if (optimize) {
    if (version >= "0.3.10") {
      // throw? or perhaps we should interpret "true" as `--optimize gas`
    }
    // for older versions there's nothing to do, because the optimizer is enabled by default
  } else {
    if (version < "0.3.10") cmd += "--no-optimize"
    else cmd += "--optimize none"
  }
} else if (typeof optimize === "string") {
  cmd += `--optimize ${optimize}`
}
```

Reply:
Right, the `--no-optimize` flag was added in version `0.3.1` and also the attribute `"optimize": true,` in the "Input JSON File" (see [here](https://github.com/vyperlang/vyper/pull/2522)). Before that Vyper had always enabled the optimiser and the same holds true until today, even though the default is `gas` now in the latest version. So taking your snippet I above, I would do the following:

```ts
if (typeof optimize === "boolean") {
  if (optimize) {
    if (version >= "0.3.10" || version < "0.3.1") {
      // throw? or perhaps we should interpret "true" as `--optimize gas`
    }
    // for older versions there's nothing to do, because the optimizer is enabled by default
  } else {
    if (version < "0.3.10") cmd += "--no-optimize"
    else cmd += "--optimize none"
  }
} else if (typeof optimize === "string") {
  cmd += `--optimize ${optimize}`
}
```